packages/plugins-image/src/components/Image.js

```diff
@@ -0,0 +1,206 @@
+    this.attemptBlobUpload();
+  }
+  componentWillUnmount() {
+    if (this.uploader) this.uploader.cancel();
+  }
+
+  onImgLoad = () => {
+    // Non-blobs onLoad should finish loading
+    if (!this.state.src.startsWith("blob:")) {
+      this.setState({ loading: false });
+    }
+  };
+
+  attemptBlobUpload() {
+    const { src } = this.state;
+    if (src && src.includes("blob")) {
+      fetch(src)
+        .then(resp => resp.blob())
+        .then(blob => {
+          const file = new File([blob], `clipboard.png`);
+          this.attemptUpload(file);
+        });
+    }
+  }
+
+  attemptUpload(file) {
+    this.setState({ loading: true });
+    const { uploadImage } = this.props.options;
+    if (uploadImage) {
+      this.uploader = makeCancelable(uploadImage(file));
+      this.uploader.promise
+        .then(newUrl => {
+          this.updateSrc(newUrl);
+        })
+        .catch(() => {
+          this.updateSrc("");
+          this.updateError(`Failed to upload file to server`);
+        });
+    } else {
+      this.setState({ loading: false });
+    }
+  }
+
+  updateError = error => {
+    this.setState({ error });
+  };
+
+  updateSrc = (src = "", record = false) => {
+    this.setState({ src, loading: false, error: "" });
+    this.props.editor.change(change => {
+      if (!record) change.setOperationFlag("save", false);
+      change.setNodeByKey(this.props.node.key, { data: { src } });
+      if (!record) change.setOperationFlag("save", true);
+    });
+  };
+
+  validFileType = type => {
+    if (validImageFormats.includes(type)) {
+      return true;
+    }
+    this.updateError("Uploaded file is not an image");
+    return false;
+  };
+
+  validFileSize = size => {
+    const defaultMaxFileSize = 10485760;
+    const maxFileSize = this.props.options.maxFileSize || defaultMaxFileSize;
+    if (size <= maxFileSize) {
+      return true;
+    }
+    this.updateError(
+      `The file exceeded the maximum size of ${bytesToMb(maxFileSize)}`
+    );
+    return false;
+  };
+
+  handleInsertImage = (event, input) => {
+    const file = event.target.files[0];
+    if (
+      !file ||
+      !this.validFileSize(file.size) ||
+      !this.validFileType(file.type)
+    ) {
+      return;
+    }
+
+    const newSrc = URL.createObjectURL(file);
+    this.updateSrc(newSrc);
+
+    this.attemptUpload(file);
+
+    resetForm(input);
+  };
+
+  deleteImage = e => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const { node, readOnly, editor } = this.props;
+    if (!readOnly) {
+      editor.change(change => {
+        change.removeNodeByKey(node.key);
+      });
+    }
+  };
+
+  selectFile = () => {
+    this.input.click();
+  };
+
+  createInput = () => (
+    <input
+      type="file"
+      accept=".gif, .jpeg, .jpg, .png, image/gif, image/jpeg, image/jpg, image/png"
+      ref={ref => {
+        this.input = ref;
+      }}
+      onChange={e => this.handleInsertImage(e, this.input)}
```

Comment:
Avoid creating an anonymous function does not need to be created here with `onChange={this.handleInsertImage}` . The reference `this.input` can be accessed from inside `this.handleInsertImage`